Rename to eforms models + Implement support for SF

duprijil opened the pull request:
The changes ads separation of eForms and SF Metadata in MSSDK + Implements support for SF

Note: package_F03 has different structure of output folder from other SF packages, so it's only package that MSSDK fails to load.

Note: A lot of file changed because new test data was added regarding SF MP:
- .zip file
- folder with MP
- pydantic model as json
All this data is used to test different parts of MSSDK

duprijil committed:
feat: Make CoreModel to serialise with the loaded alias

duprijil committed:
refactor!: Rename mapping_package module to full_mapping_package

duprijil committed:
refactor!: Rename MappingPackage -> FullMappingPackage

duprijil committed:
Revert "refactor!: Rename MappingPackage -> FullMappingPackage"

This reverts commit 93d679bc30f59d5ed5c345b2c223f50e4040b826.

duprijil committed:
Revert "refactor!: Rename mapping_package module to full_mapping_package"

This reverts commit 8e96cc1d70a71ae71d29df5524fc2d03dbb791b6.

duprijil committed:
Revert "feat: Make CoreModel to serialise with the loaded alias"

This reverts commit b97a13fd9d6ef0b6ce839bbc109b89b4c9e88fad.

duprijil committed:
feat: Separate eForms mapping package metadata with a general MP class

duprijil committed:
test: Add SF test data (model in json, folder and zip)

duprijil committed:
feat: Implement support for SF

duprijil committed:
test: Enrich tests with SF support

duprijil committed:
chore: Delete unnecessary comments from test_utils.py

costezki commented on mapping_suite_sdk/adapters/hasher.py:
```diff
@@ -62,7 +62,8 @@ def hash_mapping_metadata(self) -> str:
         list_of_hashes = self.hash_critical_mapping_files()
         signatures = [signature[1] for signature in list_of_hashes]
-        signatures.append(self.hash_mapping_metadata())
+        if self.mapping_package.metadata.type == "eforms":
```
is it necessary to have this hardcoded?

costezki commented on mapping_suite_sdk/adapters/serialiser.py:
```diff
@@ -102,9 +102,14 @@ def serialise(self, package_folder_path: Path, asset: MappingPackageMetadata) ->
         metadata_path = package_folder_path / asset.path
         metadata_path.parent.mkdir(parents=True, exist_ok=True)
         # TODO: We need somehow to store metadata file content separately in case the ident is different
+        if asset.type == "eforms":
```
can we have somehow have these values as constants somewhere rather than putting them in the code directly? And possibly this is an adapter specific to a specific version of mapping package. 
Thus, I'm not sure it is already done, but I woudld appreciate to have references to mapping package versions rather than eForms/SF.

costezki commented on mapping_suite_sdk/models/mapping_package.py:
```diff
@@ -41,12 +42,37 @@ class MappingPackageEligibilityConstraints(CoreModel):
 
-class MappingPackageMetadata(CoreModel):
-    """A class representing the metadata of a mapping package.
+class MappingPackageMetadata(CoreModel, ABC):
```
Before implementing, we need to design it first. 
The design decisions then need to be documented and then implemented. 
We also promised a JSON schema for the Metadata.

costezki commented on tests/test_data/mapping_packages/package_F22_changed/package_F22_changed/output/form_number_F22_S03/060135-2020/test_suite_report/sparql_cm_assertions.html:
do we need such files with outputs?

costezki commented on tests/test_data/mapping_packages/package_F22_changed/package_F22_changed/output/form_number_F22_S03/060135-2020/test_suite_report/sparql_validations.json:
or this sort of files? are they used in the hashing process?

schivmeister reviewed (changes requested):
As I understand there will be some more changes required, so let's wait to see those.

duprijil committed:
refactor!: Change MP model to MPv1 and MPv2

This commit changes entirely MSSDK code by changing main models to V1 and V2.
This leaded to change entire structure, adapters, services and tests

duprijil committed:
fix: Delete wrongly added shacl_result_query.rq

duprijil committed:
fix: Fix SF test data package model dump issue (used local path for metadata, not relative to package)

duprijil committed:
test: Add test coverage for mp v1 validate service

schivmeister reviewed (approved):
There are many changes due to a significant refactor, but based on an overview given by @duprijil I understood the gist.

Tested the commands and confirmed that there are no apparent regressions for eForms (now called "v2") and that there is now support for Standard Forms ("v1") package validation.

Although naming of the new Python modules with concrete implementations looks a bit weird ("mapping_package_v2"), we will have a chance to refactor that at a later time, before the development of the next version "v3" concludes.

duprijil committed:
feat: Implement config resolver

Also implement config resolver form mp v1 and v2 paths and for mssdk core configs

duprijil committed:
test: Implement unit tests for config resolver

duprijil committed:
fix: Fix problem with wrong shacl_result_query.rq path

duprijil committed:
fix: Solve problem with not loading shacl query
